autoextract/aio/client.py

```diff
@@ -77,21 +181,28 @@ async def request_raw(query: Query,
     """
     if agg_stats is None:
         agg_stats = AggStats()  # dummy stats, to simplify code
+
+    # Keep state between executions/retries
+    request_processor = RequestProcessor(query, handle_retries)
+
     post = _post_func(session)
-    post_kwargs = dict(
-        url=endpoint,
-        json=query_as_dict_list(query),
-        auth=aiohttp.BasicAuth(get_apikey(api_key)),
-        headers={'User-Agent': user_agent(aiohttp)},
-    )
+
     response_stats = []
     start_global = time.perf_counter()
 
     async def request():
         stats = ResponseStats.create(start_global)
         agg_stats.n_attempts += 1
+
+        kwargs = dict(
```

Comment:
Nipnick: `url`, `auth` and `headers` can be cached outside the method to avoid recreating them each time.

Reply:
- `endpoint` (`url`) is already an argument so it's not expected to change during runtime. 
- `auth` and `headers` were moved outside the `request` method; thanks!